### PR TITLE
fix: distinguish getters and setters in Bun.inspect()

### LINKED ISSUE #8853: JS Setter is written as `Getter` in console.log
### What version of Bun is running?

1.0.26+c75e768a6

### What platform is your computer?

Linux 6.1.66-1-lts x86_64 unknown

### What steps can reproduce the bug?

```ts
const obj = {
  a: 1,
  get getA() {
    return this.a;
  },
  set setA(n) {
    this.a = n;
  },
};

obj.setA = 2;
console.log(obj.getA);
console.log(obj);
```

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by argosphil**
### What does this PR do?

Fixes #8853.

This changes Bun.inspect to print "[Getter]", "[Setter]", or "[Getter/Setter]" for property accessors rather than always defaulting to "[Getter]".

To do so, we need to access the GetterSetter and CustomGetterSetter structures from zig, which requires expanding the bindings somewhat.

I've chosen to duplicate the code paths for GetterSetter and CustomGetterSetter to make it easier for future PRs to change them so we actually obtain references and print the getters and setters, which we can only do in the GetterSetter case.

Finally, two new tests ensure things work, though I'm not sure how to test the CustomGetterSetter case.

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests.

- [x] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I included a test for the new code, or an existing test covers it
- [x] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed
- [x] I wrote TypeScript/JavaScript tests and they pass locally (`bun-debug test test-file-name.test`)

### CODE REVIEW

File: src/bun.js/ConsoleObject.zig [hunk 1]
```diff
@@ -921,6 +921,7 @@ pub const Formatter = struct {
         Event,
 
         Getter,
+        CustomGetter,
```
Comment (gvilums) [inline src/bun.js/ConsoleObject.zig]:
```suggestion
        GetterSetter,
        CustomGetterSetter,
```
If we're making changes it probably makes sense to change these names to be more descriptive

**Review by gvilums — COMMENTED** [review]
Thanks for the contribution, this looks good overall. One small nit:

File: src/bun.js/ConsoleObject.zig (hunk 1 above)
Comment (argosphil) [reply src/bun.js/ConsoleObject.zig]: Agreed, thanks!

**Comment by paperclover** [comment]
a merge conflict but yeah this looks good

**Comment by argosphil** [comment]
> a merge conflict but yeah this looks good

Thanks for letting me know, I hadn't noticed.

**Review by gvilums — APPROVED** [review]

**Pull request merged**